lib/format.js

```diff
@@ -497,16 +568,41 @@ const styles = {
     arrayHead: cls => `${cls} [\n`,
     arrayTail: () => ']',
     arrayEntrySep: () => ',\n',
+
+    bufferChunkSize: 32,
     bufferEmpty: () => 'Buffer <>',
     bufferStart: () => 'Buffer <',
-    bufferBody: buf => buf.toString('hex').replace(/(..)/g, '$1 ').trim(),
-    bufferEnd: () => '>',
+    bufferBody: buf => buf.toString('hex').replace(/(....)/g, '$1 ').trim(),
+    bufferEnd: buf => '  ' + bufToAscii(buf) + '>',
+
+    bufferHead: () => 'Buffer <\n',
+
+    bufferKey: i => {
+      // show line numbers as offset 0x0000 through 0xffff as zero-padded hex
+      // this will wrap around if you have more than 64kb buffer, but that's
+      // (a) highly unusual for the use cases tcompare works in, and (b) fine.
+      const h = (i^0).toString(16).slice(-4)
```

Comment:
I think this function can be reduced to `bufferKey: i => (Number(i) + 0x10000).toString(16).slice(-4)` unless I'm missing something about `i^0` vs `Number(i)` in this function?

Reply:
ah, yeah, that's good.  The `Number()` isn't necessary, becuase `i` will always be a number already.